src/core/main.c

```diff
@@ -575,6 +576,18 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
                         }
                 }
 
+        } else if (proc_cmdline_key_streq(key, "systemd.watchdog_pretimeout_governor")) {
+
+                if (proc_cmdline_value_missing(key, value) || isempty(value))
+                        return 0;
```

Comment:
in this case, shouldn't we unset the previously set `arg_watchdog_pretimeout_governor` value? i.e.

```c
arg_watchdog_pretimeout_governor = mfree(arg_watchdog_pretimeout_governor);
```